examples/embeddable_examples/public/book/book_embeddable_factory.tsx

```diff
@@ -0,0 +1,128 @@
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import { BookSavedObjectAttributes, BOOK_SAVED_OBJECT } from '../../common';
+import { toMountPoint } from '../../../../src/plugins/kibana_react/public';
+import {
+  EmbeddableFactoryDefinition,
+  EmbeddableStart,
+  ErrorEmbeddable,
+  IContainer,
+  AttributeService,
+} from '../../../../src/plugins/embeddable/public';
+import {
+  BookEmbeddable,
+  BOOK_EMBEDDABLE,
+  BookEmbeddableInput,
+  BookEmbeddableOutput,
+  BookByValueInput,
+  BookByReferenceInput,
+} from './book_embeddable';
+import { CreateEditBookComponent } from './create_edit_book_component';
+import { OverlayStart } from '../../../../src/core/public';
+
+interface StartServices {
+  getAttributeService: EmbeddableStart['getAttributeService'];
+  openModal: OverlayStart['openModal'];
+}
+
+export class BookEmbeddableFactory
+  implements
+    EmbeddableFactoryDefinition<
+      BookEmbeddableInput,
+      BookEmbeddableOutput,
+      BookEmbeddable,
+      BookSavedObjectAttributes
+    > {
+  public readonly type = BOOK_EMBEDDABLE;
+  public savedObjectMetaData = {
+    name: 'Book',
+    includeFields: ['title', 'author', 'readIt'],
+    type: BOOK_SAVED_OBJECT,
+    getIconForSavedObject: () => 'pencil',
+  };
+
+  private attributeService?: AttributeService<
+    BookSavedObjectAttributes,
+    BookByValueInput,
+    BookByReferenceInput
+  >;
+
+  constructor(private getStartServices: () => Promise<StartServices>) {}
+
+  public async isEditable() {
+    return true;
+  }
+
+  public createFromSavedObject = async (
+    savedObjectId: string,
+    input: BookEmbeddableInput,
+    parent?: IContainer
+  ): Promise<BookEmbeddable | ErrorEmbeddable> => {
+    return this.create(input, parent);
+  };
+
+  public async create(input: BookEmbeddableInput, parent?: IContainer) {
+    return new BookEmbeddable(input, await this.getAttributeService(), {
+      parent,
+    });
+  }
+
+  public getDisplayName() {
+    return i18n.translate('embeddableExamples.book.displayName', {
+      defaultMessage: 'Book',
+    });
+  }
+
+  public async getExplicitInput(): Promise<Omit<BookEmbeddableInput, 'id'>> {
+    const { openModal } = await this.getStartServices();
+    return new Promise<Omit<BookEmbeddableInput, 'id'>>((resolve) => {
+      const onSave = async (attributes: BookSavedObjectAttributes, includeInLibrary: boolean) => {
```

Comment:
why can't `includeInLibrary` be called `useRefType` immediately?

Reply:
Changed it, consistency is key!